autogen/agentchat/groupchat.py

```diff
@@ -42,6 +42,18 @@ class GroupChat:
         - "manual": the next speaker is selected manually by user input.
         - "random": the next speaker is selected randomly.
         - "round_robin": the next speaker is selected in a round robin fashion, i.e., iterating in the same order as provided in `agents`.
+        - a customized speaker selection function (Callable): if passed in a function, it will be called with the following parameters:
+            custom_speaker_selection_func:
+            Parameters:
+                - last_speaker: Agent
+                    The last speaker in the group chat.
+                - groupchat: GroupChat
+                    The GroupChat object
+            Return:
+                Return one of the following:
+                1. an `Agent` class, it must be one of the agents in the group chat.
+                2. a string from ['auto', 'manual', 'random', 'round_robin'] to select a default method to use.
```

Comment:
Do we really need to consider the case where the returned value is a string in one of those options? In that case, why don't user just set `speaker_selection_method` to be that string?

Reply:
If setting `speaker_selection_method ` to be a string, the user only uses one built-in method. The intention here is that user can select different built-in method at different scenarios. For example, always call executor after detecting code, while returning `auto` for auto selection for other cases.